### ISSUE TITLE
tf.contrib.rnn.LSTMCell wrong documentation and unclear naming  

### ISSUE THREAD

**Issue by oldsqlwnb** [tensorflow/tensorflow#20050]
### Describe the problem
1.
In the documentation at [https://www.tensorflow.org/api_docs/python/tf/contrib/rnn/LSTMCell](url) it says: 

> Long short-term memory unit (LSTM) recurrent network cell. The default non-peephole implementation is based on: http://www.bioinf.jku.at/publications/older/2604.pdf S. Hochreiter and J. Schmidhuber. "Long Short-Term Memory". Neural Computation, 9(8):1735-1780, 1997

This is not true since the implementation already contains the forget gate which was not mentioned in the original paper but in this one: 

> "Learning to Forget: Continual Prediction with LSTM" by Gers et al.

2.
The naming of the parameter `num_units` or the `LSTMcell` is  is unclear and causes a lot of confusion. (google "num_units lstm cell" and you will find countless posts with the same question: what does it mean?)

Since the number of units within an LSTM cell is fixed, `num_units` should either `memory_cell_block_size` as it is described in the "Long Short-Term Memory" paper and/or rename `LSTMcell` into `LSTMMemoryCellBlock`

### Request
1. Add the Paper:
> "Learning to Forget: Continual Prediction with LSTM" by Gers et al.

 to the Documention and mention that the implementation already contains forget gates.

2. Rename `num_units` and/or  `LSTMcell` or at the very least add a comment

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by oldsqlwnb** [comment]
I don't see how those fields are relevant, since it is more a documentation problem.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @michaelisard: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 80 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 110 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by MarkDaoust** [comment]
Your suggestions sound like a pretty concrete proposal for some doc-updates. Any chance you can submit them as a PR?

It's too late to change the name if num_units parameter.
But clarifying the description in the doc-string could be helpful.

**Comment by oldsqlwnb** [comment]
I see that the correct paper has been referenced. I will see if I can find some time to issue and PR, regarding the num_units parameter.

**Comment by mohantym** [comment]
Is this still an issue? Have you checked latest [LSTMcelldocument](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LSTMCell) from 2.7 yet?
